### PR TITLE
added authentication via cookies

### PR DESCRIPTION
**Pull request by knavesec**
Been getting annoyed with the LinkedIn login not working consistently so added the ability to authenticate to LI via session cookies. 

Process:
1. Log into LinkedIn via a browser
2. Copy the `li_at` cookie
3. Specify as `python3 linkedin2username.py --cookie AQEDASA_lBgDx1DqAAABihtN.......` and it should authenticate you properly

Feel free to update the documentation

### CODE REVIEW

File: linkedin2username.py [hunk 1]
```diff
@@ -253,7 +256,8 @@ def parse_arguments():
 
     # If password is not passed in the command line, prompt for it
     # in a more secure fashion (not shown on screen)
-    args.password = args.password or getpass.getpass()
+    if args.cookie is None: 
+        rgs.password = args.password or getpass.getpass()
```
Comment (initstring) [inline linkedin2username.py]:
```suggestion
        args.password = args.password or getpass.getpass()
```

File: linkedin2username.py [hunk 2]
```diff
@@ -313,6 +317,14 @@ def login(args):
         'loginCsrfParam': login_csrf
         }
 
+    # If authentication via cookies is specified, set the important cookie 
+    # and return the session
+    if args.cookie is not None:
```
Comment (initstring) [inline linkedin2username.py]:
```suggestion
    if args.cookie:
```

**Comment by initstring** [comment]
Hi @knavesec ,

Thank you for the nice PR. This is a good idea, I will merge it in. I may tweak it a bit first, to use a mutually exclusive arg group between username and cookie.

Thanks again!

**Comment by initstring** [comment]
Hi @knavesec - I was a bit overzealous in merging this. Sorry about that, but I had to revert it.
It wouldn't work for me, generating 403 errors when attempting to get the company info. Was it working for you?

**Comment by initstring** [comment]
Hi @knavesec - just FYI, I completely abandoned the CLI-powered auth. It's just too complex to keep it working.

The newest version in my branch uses Selenium to handle auth, which solves all this.

Thanks again for your PR, it inspired me to try some new things. :)

**Comment by knavesec** [comment]
@initstring  Interesting that it didn't work, never failed for me! Either way, glad we got a solution working, great job and great tool!

**Pull request merged**